AnkiDroid/src/main/java/com/ichi2/libanki/Collection.java

```diff
@@ -330,11 +330,9 @@ private int getChunk() {
         // We have the ability to look into our sqlite implementation on Android and use it's value
         // as a ceiling. Try it, with a reasonable fallback in case of failure
         SupportSQLiteDatabase db = mDb.getDatabase();
-        if (! (db instanceof DatabaseChangeDecorator)) {
-            return sChunk;
-        }
-        String db_name = ((DatabaseChangeDecorator) db).getWrapped().getClass().getName();
-        if ("io.requery.android.database.sqlite.SQLiteDatabase".equals(db_name)) {
+        String db_name = (db instanceof DatabaseChangeDecorator) ? ((DatabaseChangeDecorator) db).getWrapped().getClass().getName() : null;
+
+        if (db_name != null && "io.requery.android.database.sqlite.SQLiteDatabase".equals(db_name)) {
```

Comment:
Because we use the style of "constant string first in string comparison" you don't need the null check (that is the whole point of the constant string first actually: it avoids null pointers) so you can do it like this:

```suggestion
        if ("io.requery.android.database.sqlite.SQLiteDatabase".equals(db_name)) {
```

Reply:
Yes, this looks good. I added the null check as I thought it'd give a NullPointerException. But now I'm clear that it won't.

Reply:
Thanks for the suggestion. It's been a good learning so far contributing to AnkiDroid :)